### ISSUE TITLE
Need to make sure that `arbitrary-users-patch` images work on OpenShift v4

### ISSUE THREAD

**Issue by ibuziuk** [eclipse-che/che#13960]
We have to do `arbitrary-users-patch` [1] of the community images been supported on OpenShift. During the discussion on https://github.com/openshift/origin/issues/23369 it appeared to be that `CRI-O now has support for adding the entry of the user to the /etc/password if it does not exists in /etc/passwd.` Assuming that CRI-O is the default contianer engine for OpenShift v4 we need to verify that our patched images work correctly on both OpenShift v3 and OpenShift v4 and there are no side-effects.

[1] https://github.com/eclipse/che-devfile-registry/tree/master/arbitrary-users-patch

**Comment by ibuziuk** [comment]
@rhopp should help us with the OpenShift v4 access

**Comment by rhopp** [comment]
I have Che 7 (some nightly version) deployed on OCP 4 right now (while trying something else), so I tried to start workspace from this devfile: https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/java-maven/devfile.yaml

Image used in the devfile:
```
Image:         quay.io/eclipse/che-java11-maven:nightly
    Image ID:      quay.io/eclipse/che-java11-maven@sha256:33000ee0000b18ea5a7dd8c32cc4871ed7c5baba03c449f0110f5b33e06718c9
```
This is how it looks like in Che, when I open terminal:
![image](https://user-images.githubusercontent.com/1215011/61708808-fe2d9d00-ad4d-11e9-9886-3e688941d575.png)
@amisevsk This doesn't look correct, right?  ^^

**Comment by amisevsk** [comment]
Nope, the entrypoint doesn't seem to run here.

**Comment by ibuziuk** [comment]
@rhopp @l0rd @slemeur please consider adding this issue to the end game plan during triage (I think we need to add it as a subtask for Verify and fix devfile registry images to work on openshift )

**Comment by ibuziuk** [comment]
@rhopp `quay.io/eclipse/che-java11-maven:nightly` definitely works on `v3.11.82` it is pretty frustrating if this is failing on OpenShift v4

**Comment by l0rd** [comment]
@rhopp are you able to get the UID and the corresponding line in the /etc/passwd in the maven container?

**Comment by l0rd** [comment]
That is pretty weird. Assuming `CRI-O` updates `/etc/passwd`:
1) [you should not be able to open a terminal ](https://github.com/eclipse/che/issues/13977) but apparently you can (maybe in your case it has used another default login shell?)
2) [entrypoint.sh](https://github.com/eclipse/che-devfile-registry/blob/master/arbitrary-users-patch/entrypoint.sh) should still be executed but it looks like it's not (maybe it failed and return before completing?)

**Comment by rhopp** [comment]
@l0rd We've found the reason for my failure - I was using old devfile, which has "command" for the container defined -> thus the pod has "command" defined meaning openshift overriden entrypoint with this command.

**Comment by ibuziuk** [comment]
Have tested with the following devfiles form the registry against OCP v4 and was not able to spot any issues:
 - `java-maven` - https://github.com/eclipse/che-devfile-registry/tree/master/devfiles/java-maven
 - `go` - https://raw.githubusercontent.com/eclipse/che-devfile-registry/master/devfiles/go/devfile.yaml
 - `python` - https://raw.githubusercontent.com/eclipse/che-devfile-registry/master/devfiles/python/devfile.yaml
- `php-web-simple` - https://raw.githubusercontent.com/eclipse/che-devfile-registry/master/devfiles/php-web-simple/devfile.yaml

Basically, all the workspaces started from those devfiles have a dedicated `/etc/passwd` entry and `whoami` return `user`, witt `id` like uid=1000820000(user) gid=0(root) groups=0(root),1000820000

@rhopp @l0rd I believe we can close this issue as done.

**Comment by l0rd** [comment]
@ibuziuk cool!
